daemon/cmd/fqdn_test.go

```diff
@@ -92,21 +114,22 @@ func BenchmarkNotifyOnDNSMsg(b *testing.B) {
 	)
 
 	// Register rules (simulates applied policies).
+	dscu := &dummySelectorCacheUser{}
 	selectorsToAdd := api.FQDNSelectorSlice{ciliumIOSel, ciliumIOSelMatchPattern, ebpfIOSel}
-	nameManager.Lock()
 	for _, sel := range selectorsToAdd {
-		nameManager.RegisterForIPUpdatesLocked(sel)
```

Comment:
Just curious, why are we not adding the fqdnselector via `namemanager` and directly though selector cache ?